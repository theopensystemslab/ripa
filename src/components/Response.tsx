import Grid from "@material-ui/core/Grid";
import React from "react";

import ResponseButton from "./ResponseButton";
import ResponseImageButton from "./ResponseImageButton";

const Response = ({
  selected = null,
  handleClick = null,
  response,
  responseKey = []
}) => {
  const responseChar =
    responseKey && responseKey.map(c => String.fromCharCode(c)).join("");
  if (response.img) {
    return (
      <Grid item xs={6} sm={4} lg={3}>
        <ResponseImageButton
          key={response.id}
          selected={selected}
          handleClick={handleClick}
          responseKey={responseChar}
          image={response.img && <img src={response.img} />}
        >
          {response.text}
        </ResponseImageButton>
      </Grid>
    );
  } else {
    return (
      <Grid item xs={12}>
        <ResponseButton
          key={response.id}
          selected={selected}
          handleClick={handleClick}
          responseKey={responseChar}
        >
          {response.text}
        </ResponseButton>
      </Grid>
    );
  }
};
export default Response;
