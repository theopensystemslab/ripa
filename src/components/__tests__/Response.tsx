import { cleanup, fireEvent, render, wait } from "@testing-library/react";
import * as React from "react";

import Response from "../Response";

afterEach(cleanup);

describe("Response Component", () => {
  it("should render snapshot when image is available", () => {
    const response = { id: "1", text: "example text", img: "img.png" };
    const { asFragment } = render(<Response response={response} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("should render snapshot when image is not available", () => {
    const response = { id: "2", text: "example text" };

    const { asFragment } = render(<Response response={response} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
