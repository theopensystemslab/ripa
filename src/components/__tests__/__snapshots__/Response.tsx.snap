// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Response Component should render snapshot when image is available 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 MuiGrid-grid-lg-3"
  >
    <button
      class="MuiButtonBase-root makeStyles-root-1"
      tabindex="0"
      type="button"
    >
      <div
        class="makeStyles-buttonContent-8"
      >
        <div
          class="makeStyles-imageContainer-2"
        >
          <img
            src="img.png"
          />
        </div>
        <div
          class="MuiGrid-root makeStyles-text-6 MuiGrid-container MuiGrid-wrap-xs-nowrap"
        >
          <div
            class="MuiGrid-root makeStyles-key-5 MuiGrid-item"
          >
            <p
              class="MuiTypography-root makeStyles-type-7 MuiTypography-body1 MuiTypography-colorInherit"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <p
              class="MuiTypography-root makeStyles-type-7 MuiTypography-body1 MuiTypography-colorInherit"
            >
              example text
            </p>
          </div>
        </div>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;

exports[`Response Component should render snapshot when image is not available 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <button
      class="MuiButtonBase-root makeStyles-root-9"
      tabindex="0"
      type="button"
    >
      <div
        class="MuiBox-root MuiBox-root-13 makeStyles-key-10"
      />
      <div
        class="MuiBox-root MuiBox-root-14 makeStyles-text-12"
      >
        example text
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
